refract-spec.md

```diff
@@ -137,9 +133,419 @@ This is how it would expressed in the compact version.
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Example
+
+```json
+{
+  "element": "string",
+  "attributes": {
+    "id": "foo"
+  },
+  "content": "bar"
+}
+```
+
+### Ref (Element)
+
+The `ref` element can be used to reference elements in remove documents or elements in the local document.
+
+1. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+1. If the content is a relative fragment identifier, it references an element found in the local document
+1. If there is no fragment identifier, the reference is to the entire document.
+1. If there are multiple elements in a document with the same ID, the first element with the matching value SHOULD be used.
+
+The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL
+
+#### Examples
+
+Elements can be referenced in remove or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. Namespaces do not include the elements and the data in the elements.
+
+### Namespaces (Element)
+
+The `namespaces` element is a list of namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
+
+#### Members
+
+- (string) - A resolvable URL to a namespace
+- (object)
+    - `prefix` (string) - A prefix
+    - `href` (string) - An `href` is a resolvable URL to a namespace
+
+#### Example Using Both String and Object Namespace Elements
+
+This example shows a namespace being included by way of a string, and a prefixed namespace.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "namespaces": [
+      "http://example.com/namespace1",
+      { "prefix": "ns2", "href": "http://example.com/namespace2" }
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "prefix": "ns2"
+  },
+  "content": "baz"
+}
+```
+
+## Extending Elements
+
+Elements by default do not include any of the attributes or content from the element in which they reference. An element may be extended in order to inherit from other elements.
+
+### Extend (Element)
+
+The `extend` element provides a way to merge one or more elements into a new element. The `extend` element MUST NOT affect the elements being extended, but MUST define a new instance of an element.
```

Comment:
How about:

The `extend` element provides a way to multiply inherit from one or more elements to from a new element. The `extend` element MUST NOT modify the elements being extended, but MUST define a new element.

I took out instance. Since this is defining an element type, I don't think instance is right (even though the result is an element). I think an instance refers more to using the type in an element per the structure not pulling in properties, etc. Make sense?